blockchain/src/blockchain/blockchain.rs

```diff
@@ -34,17 +32,15 @@ pub struct Blockchain {
     // The OffsetTime struct. It allows us to query the current time.
     pub time: Arc<OffsetTime>,
     // The notifier processes events relative to the blockchain.
-    pub notifier: RwLock<Notifier<'static, BlockchainEvent>>,
+    pub notifier: Notifier<'static, BlockchainEvent>,
     // The fork notifier processes fork events.
-    pub fork_notifier: RwLock<Notifier<'static, ForkEvent>>,
+    pub fork_notifier: Notifier<'static, ForkEvent>,
     // The chain store is a database containing all of the chain infos, blocks and receipts.
-    pub chain_store: Arc<ChainStore>,
+    pub chain_store: ChainStore,
     // The history store is a database containing all of the history trees and transactions.
-    pub history_store: Arc<HistoryStore>,
+    pub history_store: HistoryStore,
     // The current state of the blockchain.
-    pub state: RwLock<BlockchainState>,
-    // A write lock for the blockchain. Guarantees that only one thread writes to it at a time.
-    pub(crate) push_lock: Mutex<()>,
+    pub state: BlockchainState,
```

Comment:
Actually, the BlockchainState was only needed for the locking. So, now we could simply remove it. But I'm equally fine with keeping it as it encapsulates the state quite nicely.

Reply:
I think keeping it makes sense as you can easily borrow a reference to it.